command_runner/runner.go

```diff
@@ -29,14 +30,22 @@ type Runner interface {
 	RunSubCmdByName(cmdName string, subCommand string, c *cli.Context) (err error)
 }
 
+type Factory interface {
```

Comment:
Why copy the interface here?  The code actually now has three copies of this, in `factory.go`, `runner.go` and `rpc_server.go`.   Surely it's fine for `factory.go` to declare it and have other code use it?  Same thing applies to `Runner` interface.